src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java

```diff
@@ -0,0 +1,209 @@
+    this.config = config;
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateConnectionParameters(collector);
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) {
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    List<BrowseEntity> entityList = Arrays.stream(ObjectType.values()).filter(t -> t.isSamplingAllowed())
+      .map(object -> {
+        BrowseEntity.Builder entity = (BrowseEntity.builder(object.getObjectName(),
+          object.getObjectName(),
+          ENTITY_TYPE_OBJECT).canBrowse(false).canSample(true));
+        return entity.build();
+      }).collect(Collectors.toList());
+    return browseDetailBuilder.addEntities(entityList).setTotalCount(entityList.size()).build();
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest) {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
+    Map<String, String> properties = new HashMap<>();
+    properties.put(ConfigUtil.NAME_USE_CONNECTION, "true");
+    properties.put(ConfigUtil.NAME_CONNECTION, connectorSpecRequest.getConnectionWithMacro());
+    String objectName = connectorSpecRequest.getPath();
+    if (objectName != null) {
+      properties.put(ZendeskBatchSourceConfig.PROPERTY_OBJECTS_TO_PULL, objectName);
+      properties.put(Constants.Reference.REFERENCE_NAME, ReferenceNames.cleanseReferenceName(objectName));
+    }
+    Schema schema = ObjectType.fromString(objectName, collector).getObjectSchema();
+    specBuilder.setSchema(schema);
+    return specBuilder.addRelatedPlugin(
+      new PluginSpec(ZendeskBatchSource.NAME, BatchSource.PLUGIN_TYPE, properties)).build();
+  }
+
+  @Override
+  public List<StructuredRecord> sample(ConnectorContext connectorContext, SampleRequest sampleRequest)
+    throws IOException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    List<StructuredRecord> recordList = new ArrayList<>();
+    objectType = ObjectType.fromString(sampleRequest.getPath(), collector);
+    Schema schema = objectType.getObjectSchema();
+    for (String record : getResponseAsMap()) {
+      recordList.add(StructuredRecordStringConverter.fromJsonString(record, schema));
+    }
+    return recordList;
+  }
+
+  private List<String> getResponseAsMap() throws IOException {
+    subdomain = config.getSubdomains().stream().findFirst().get();
+    String baseUrl = String.format(ZendeskConstants.BASE_URL, subdomain, objectType.getApiEndpoint());
+    if (objectType.isBatch()) {
+      // start time = current date - 30 days
+      // start time is set to 30 days as only sample data is needed which can be fetched from the data of particular
```

Comment:
How did you choose 30 days? Isn't it possible that there might be a huge amount of data in the last 30 days?
Instead, is it possible to get the last N records where N = sampleRequest.getLimit() ?